tree/tree/src/TTree.cxx

```diff
@@ -4383,53 +4383,56 @@ Int_t TTree::Fill()
 {
    Int_t nbytes = 0;
    Int_t nerror = 0;
-   Int_t nb = fBranches.GetEntriesFast();
-   if (nb == 1) {
-      // Case of one single super branch. Automatically update
-      // all the branch addresses if a new object was created.
-      TBranch* branch = (TBranch*) fBranches.UncheckedAt(0);
-      branch->UpdateAddress();
-   }
-   if (fBranchRef) {
+   Int_t nbranches = fBranches.GetEntriesFast();
```

Comment:
Sure, I will add explanations, as this is actually quite complicated to parse. There was a lot of branching in the previous code that I tried to reduce, since this is a very performance-critical function. New commit message coming, also with a link to the [JIRA issue](https://sft.its.cern.ch/jira/browse/ROOT-8872) where the analysis is being documented.